pkg/agent/openflow/multicast.go

```diff
@@ -112,6 +118,30 @@ func (f *featureMulticast) multicastReceiversGroup(groupID binding.GroupIDType,
 	return group
 }
 
+// multicastForwardInboundGroup forwards ingress multicast traffic to host interface
+// and multicast pipeline when flexibleIPAM is enabled.
+func (f *featureMulticast) multicastForwardInboundGroup(groupID binding.GroupIDType, tableID uint8) binding.Group {
+	group := f.bridge.NewGroupTypeAll(groupID)
+
+	group = group.Bucket().
+		ResubmitToTable(tableID).
+		Done()
+	group = group.Bucket().Output(f.hostOFPort).Done()
+	return group
+}
+
+// multicastForwardOutboundGroup forwards ingress multicast traffic to uplink interface
+// and multicast pipeline when flexibleIPAM is enabled.
+func (f *featureMulticast) multicastForwardOutboundGroup(groupID binding.GroupIDType, tableID uint8) binding.Group {
+	group := f.bridge.NewGroupTypeAll(groupID)
+
+	group = group.Bucket().
+		ResubmitToTable(tableID).
+		Done()
+	group = group.Bucket().Output(f.uplinkPort).Done()
```

Comment:
Does it cause the multicast packet duplicated, one is output to uplink via line 141, the second is from the group after matching MutlicastRoutingTable via line 139?